lib/compiler/src/compile.erl

```diff
@@ -154,6 +154,44 @@ other Erlang code.
 See `m:erl_id_trans` for an example and an explanation of the function
 `parse_transform_info/0`.
 
+## Recommendations for Language Implementors
+
+Except for Erlang, there are quite a few other languges that can run
+on the Erlang runtime system. Generally, such compilers produce a file
+or binary that is fed into the Erlang compiler. There are four ways
+to do that using documented functionality:
+
+* **Erlang source code (`.erl` file)**: The file is compiled
+  using `compile:file/2` or `erlc <FILE>.erl`. This is the most
+  straightforward and portable way. The main disadvantage is that
+  it is hard or impossible to map each line of Erlang code back to the
+  corresponding line in the original source file.
+
+* **[The abstract format](`e:erts:absform.html`)**: This format
+  supports mapping every Erlang source line back to its corresponding
+  line in the original source file. To compile from the abstract
+  format, do one following:
+
+    * Call `compile:forms/2` to directly pass in the term
+      representation of the [abstract format](`e:erts:absform.html`).
+
+    * Write the term representation to a file with the extension
+      `.abstr` and compile that file using `compile:file(<File>,
+      [from_abstr])` or `erlc <File>.abstr`.
+
+* **Core Erlang**: While there is a specification for Core Erlang,
+  certain details are left to the implementation via the `primop`
+  expression. Primops can be added, deleted, or changed in any major
+  release without notice.
```

Comment:
Maybe add something like "Note that by generating Core Erlang directly, it is possible to construct code that the Core-to-BEAM backend has never encountered before, and there are no guarantees that the final BEAM code will be safe."